### PR TITLE
record: 10L d496 WarmDown3500 SWA — val_bpb 1.1590 (1xH100 proxy)

### PR DESCRIPTION
**Pull request by Hilo-Hilo**
## 10L d496 WarmDown3500 + SWA (env-only tuning)

**val_bpb: 1.1590** (int8+zlib roundtrip) | **15.94 MB** | 1xH100 NVL proxy

### Honesty up front

This was trained on a **single H100 NVL**, not the official 8xH100 setup. It reached 6,721 steps in 5,100s (vs ~13,000+ steps in 600s on 8xH100). We have not yet verified the 8xH100 result but expect it to be somewhat better. Submitting to share the configuration — happy to update with 8xH100 numbers if someone can repro.

### Approach

No code changes at all. Stock `train_gpt.py` with env overrides:

```bash
MODEL_DIM=496 NUM_LAYERS=10 WARMDOWN_ITERS=3500 \
SWA_ENABLED=1 SWA_START_FRAC=0.4 SWA_EVERY=50 \
EVAL_STRIDE=64 TTT_ENABLED=0 \
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

Key choices:
- **d_model=496** (not 512) to fit under 16MB after int6+zlib
- **warmdown=3500** (not 3000) for slightly better convergence
- **TTT disabled** — keeps eval fast; re-enabling with defaults would likely improve by 0.002-0.005

### Limitations

- Single seed, no significance testing
- 1-GPU proxy only (46% of 8xH100 training steps)
- No novel techniques — purely hyperparameter tuning
- TTT disabled (easy improvement left on the table)

### Results (1xH100 NVL proxy)

| Metric | Value |
|--------|-------|
| exact_final_val_bpb | **1.1590** |
| pre_quant_val_bpb | 1.1695 |
| steps | 6,721 |
| bytes_total | 15,940,324 |

This was found by an autonomous research swarm (Claude agents planning and running experiments on RunPod H100s). The swarm ran ~64 experiments over ~18 hours and this was the 2nd best result.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — record: 10L d496 WarmDown3500 SWA — val_bpb 1.1590 (1xH100 proxy)

**Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

--- ## Analysis ### 1. ILLEGAL n-gram family bug — NOT PRESENT No n-gram, bigram, BigramHash, or hash-key XOR patterns found anywhere in the file. The only `^` occurrence is in a comment (line 607: `relu^2`). No target-token XOR'd into a hash key. ### 2. ILLEGAL Pre-Quant TTT (multi-epoch on val_tokens) — NOT PRESENT No TTT, test-time training, or fine-tuning loop over `val_tokens` found. `val_tokens` is used exclusively in `eval_val()` under `torch.inference_mode()` (line 250), which is called from the standard validation check (lines 976-997) and the post-quantization roundtrip check (lines 1102-1113). No backward pass is ever called against `val_tokens`. ### 3. LEGAL score-first TTT (PR #1413 pattern, is_last_chunk guard) — NOT PRESENT No TTT of any kind. No `is_last_chunk` guard, no score-first pattern. Clean. ### 4. HOLD scored-region SLOT — NOT PRESENT No scored-region slot usage detected. ### 5. Architecture summary - 10-layer GPT with U-Net style skip connections (encoder/decoder split with `skip_weights`, lines 670-713) - GQA attention (8 heads, 4 KV heads), RoPE, logit softcap (tanh) - relu^2 MLP, RMSNorm, CastedLinear, Muon + Adam optimizer split - WarmDown LR schedule (WARMDOWN_ITERS=3500 per directory name) - Standard wallclock-capped training loop; no SWA despite the directory name suggesting it (no averaging code found) - Post-training int8 + zlib quantization with roundtrip validation - Tied embeddings with vocab_size=1024 This is a well-engineered pure neural submission. The "SWA" in the directory name is not backed by any stochastic weight averaging code in the file — the directory name appears aspirational or mislabeled. No...

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the audit — this looks like a clean pure-neural submission.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). Compliance audit via LLM agent (Sonnet) reviewing full train_gpt.py source, cross-checked against deterministic AST classifier. If this review misread your code, please call it out so I can re-audit manually.*
